pkg/identity/api/v1/service.go

```diff
@@ -41,14 +41,45 @@ func (s *Service) Close() {
 	s.log.Info("closed")
 }
 
+/*
+Algorithm:
+
+Start transaction
+
+ 1. Insert the update into the inbox_log table
+ 2. Read the log for the inbox_id, ordering by sequence_id
+ 3. If the log has more than 256 entries, abort the transaction.
+ 3. Validate it sequentially. If failed, abort the transaction.
+ 4. For each affected address:
+    a. Insert the update into the address_log table
+    -- Note: There may be races across multiple inboxes. The address_log can use
+    -- inbox_log_sequence_ID to establish ordering.
+    b. Read the log for the address, ordering by *inbox_log_sequence_id*
+    c. If the log has more than 256 entries, abort the transaction.
+    c. Process the address log (without signature validation) and compute the final inbox ID
```

Comment:
There's really only two things we need to check here:
- That there are < 256 updates in the log
- That the log is either empty or the last message is a revocation

We might be able to get away with just a `COUNT` query and pulling the last record, and skip all the middle entries.

Reply:
I'm thinking about this scenario:

```
Associate address to inbox 1
Associate address to inbox 2
Revoke address from inbox 1
```

Reply:
It's not so much verifying, as figuring out what the correct inbox ID should be based on the entries

Reply:
Ohhh, I see. What I proposed only works for adds but doesn't help at all for revocation

Reply:
Thinking about the revocation case a bit more.

What about this model:

```sql
CREATE TABLE address_log (
    associated_at_inbox_log_sequence_id BIGINT NOT NULL,
    revoked_at_inbox_log_sequence_id BIGINT,
    address TEXT NOT NULL,
    inbox_id TEXT,
);
```

When a revocation happens we just update the previous association by setting the `revoked_at_inbox_log_sequence_id` field.

We can then query it by only looking for results where the `revoked_at_inbox_log_sequence_id` field is null and get the last still-valid association.

Reply:
Just avoids the need to re-process the address log on updates, which seems complicated since the individual entries only make sense in the context of an `identity_log` history

Reply:
I don't love that this is still not purely append-only, which would be the easiest to decentralize.

Reply:
I like this, it's more scalable. I generalized it a little to handle cases where there are multiple associations and revocations